skip-file.php

```diff
@@ -97,3 +97,104 @@ function vipgoci_get_skipped_files_issue_message(
 		$affected_files
 	);
 }
+
+/**
+  * IF the file
+ *  Is modified in THIS very commit
+ *  Or New in THIS very commit
+ *  Or Doesn't exist in the previous comments $reached_limit_files
+ *      Let the bot post the comment
+ * ELSE:
+ *     REMOVE FROM the $results
+ *
+ * @param array $pr_issues_results
+ * @param array $comments
+ * @param array $modified_files
+ */
+function vipgo_skip_file_check_pr_comments(array $pr_issues_results = [], array $comments = [], array $modified_files = [])
+{
+	// git diff-tree --no-commit-id --name-only -r 773b2bc0b46d0cf6fde031af79e957d3a0cc9172
+	/**
+	 * If there is no PR previous comments, return
+	 */
+	if ( 0 === count( $comments ) ) {
+		return $pr_issues_results;
+	}
+
+	/**
+	 * List of modified files in this very COMMIT
+	 * @todo get_list
+	 */
+	$reached_limit_files = [];
+
+	/**
+	 * Iterates all the comments to check the files affected by the skip files due max lines limit reached
+	 * @todo: move the comment iteration logic to its own function
+	 */
+	foreach ( $comments as $comment ) {
+		/**
+		 * @todo use options limit values and constant
+		 *  sprintf(VIPGOCI_VALIDATION[VIPGOCI_VALIDATION_MAXIMUM_LINES], $pr_number, $option  limit)
+		 * $prefix = Maximum number of lines exceeded (15000):
+		 * $suffix = PHP_EOL . PHP_EOL . VIPGOCI_VALIDATION_MAXIMUM_DETAIL_MSG
+		 * @todo: move the getting files logic to its own function
+		 */
+		$prefix = '):';
+		$suffix = strlen( PHP_EOL . PHP_EOL . VIPGOCI_VALIDATION_MAXIMUM_DETAIL_MSG );
+		$files = explode(
+			"\n - ",
+			substr(
+				$comment->body,
+				strpos( $comment->body, $prefix ) + 6,
+				-$suffix
+			),
+		);
+
+		$reached_limit_files = array_merge( $reached_limit_files, $files );
+	}
+
+	/**
+	 * This could also be a foreach
+	 * @todo cleanup: this is the only bit that will stay in this function
+	 *
+	 * Existe na lista novo e nao tem na lista de comentarios anterioes!
+	 * Existe na lista nova e existe na lista de comentarios anteriores
+	 *      Faz parte de modificado?
+	 *          SIM: mantem
+     *          NAO: REMOVE
+	 */
+	$total_issues = $pr_issues_results['total'];
+	for ( $i = 0; $i < $total_issues; $i++ ) {
+		$file = $pr_issues_results['issues']['max-lines'][ $i ];
+
+		/**
+		 * Verifies if there's a message about skipped files in previous PR comments.
+		 * If there's not, keep the message to be posted
+		 */
+		if ( ! in_array( $file, $reached_limit_files ) ) {
+			continue;
+		}
+
+		/**
+		 * If there is: Verify if it's modified in this very commit
```

Comment:
I would avoid logic which relies on _every_ commit being scanned, as I think would be the case here. The reason is that we do not really scan every single commit in practice. Instead we should rely on all changes being scanned _eventually_.

Is this check needed to ensure not posting twice about files skipped?  It seems to me that the check above is enough? 

We may sometimes have a message being posted twice for a particular file, but if the probability is small enough, that should suffice, I think.

Reply:
This check is to avoid posting it twice, yes.

Yes, the problem is small, I just wanted to cover one more scenario. I'm going to remove this logic then. If we find that it's necessary for the future, we can add it back. 

---
Also, as a matter of understanding, could you elaborate a little more on this?

> The reason is that we do not really scan every single commit in practice

Reply:
> > Also, as a matter of understanding, could you elaborate a little more on this?

>    The reason is that we do not really scan every single commit in practice

Of course, no problem!

It may be simplest to explain this by giving an example: Let's imagine there is a pull request with one commit which has been analysed by `vip-go-ci`. Then two more commits are pushed to the pull request. While we could in theory scan _each_ of these commits, we might end up analysing commits with obsolete code, and thereby possiblity finding issues that have been fixed already (in the latest commit). This is why `vip-go-ci` should generally be constructed so that all changes are scanned _eventually_ -- it will eventually find all issues in the code. In the example provided, we would end up analysing only the latest, most current commit, finding any possible issues,  no matter if they were introduced in the first, second or third commit.

I hope this makes sense :slightly_smiling_face: